### DISCUSSION TITLE
Welcome to MaiPureSpaController Discussions!

### DISCUSSION THREAD

**Discussion by UlrichMai** [UlrichMai/MaiPureSpaController#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by katfreak2** [comment]
Hi,

I have some questions about your code.

You have defined a table with all the bits for the (select, led, 7-segment, button query).
I see that you sample each bit until you have 16bits received and than execute the next code:

void ICACHE_RAM_ATTR SPI_handleLatch() {

      if (clkCount == 16) {
        //Valid if 16 clock cycles detected since last latch
        if (false);
        else if (simulateButtonPress());
        else if (bitRead(buf, 6) == 0)  readSegment(0);
        else if (bitRead(buf, 5) == 0)  readSegment(1);
        else if (bitRead(buf, 11)== 0)  readSegment(2);
        else if (bitRead(buf, 2) == 0)  readSegment(3);
        else if (bitRead(buf, 14)== 0)  readLEDStates();
      }
      // reset buffer
      buf = 0;
      clkCount = 0;
} 

if we for example received data to run this line:  else if (bitRead(buf, 6) == 0)  readSegment(0);

readSegment(0) is directly executed so you use the same data in buf but now you are checking the 7 segment bits.
 
how is this possible? don't you need to read again 16bits before running readSegment(0)??
how do you know you received the 7 segment packet or another one?
